### DISCUSSION TITLE
Test aborts after 10 minutes

### DISCUSSION THREAD

**Discussion by FabianOswald-WolfSystem** [ricaun-io/RevitTest#14] [Q&A] [answered]
Hello, I have a test that takes about 1 hour. I can start it and Revit works. But after 10 minutes the test is aborted with the following message: "No test is available in "C:\.......Tests.dll". Make sure that the test discoverers and executors are registered and the version settings for platform and framework are correct and repeat the process."

Does RevitTest set a timeout?

**Comment by ricaun** [comment]
How many tests are you running just for curiosity. 

This 10 minutes timeout is hard coded (`10 * 60`), I guess I forgot to expose that option.

I need to fix that, I could create a `alpha` version with a bigger time just enable run with long tests before I implement some configuration like `NUnit.Timeout`.

**Reply by ricaun** [reply]
The release 1.3.7-alpha have a 120 minutes timeout.
https://www.nuget.org/packages/ricaun.RevitTest.TestAdapter/1.3.7-alpha

**Reply by ricaun** [reply, marked as answer]
The version `1.4.*` support  `NUnit.Timeout` to change the timeout value.
```C#
[assembly: AssemblyMetadata("NUnit.Timeout", "120")] // Set timeout 120 minutes, default 10 minutes
```
https://www.nuget.org/packages/ricaun.RevitTest.TestAdapter/1.4.1

**Comment by FabianOswald-WolfSystem** [comment]
Only two. 
That would be great if you could create an alpha version. Thank you very much.

**Reply by ricaun** [reply]
Why that take so long to run that test?

**Reply by FabianOswald-WolfSystem** [reply]
Because we open more than 1300 Revit families (updating them in the process), edit them and then save them again.

**Reply by ricaun** [reply]
Nice! Are you open each file in the folder or something?

I don't thing I have a public sample to update families, need to create one.

**Comment by ricaun** [comment]
I gonna release a `1.3.7-alpha` with `120 minutes`. 
The github action is running, just check in the nuget to see if is available. 
https://www.nuget.org/packages/ricaun.RevitTest.TestAdapter/1.3.7-alpha

**Comment by ricaun** [comment]
The version `1.4.0` support  `NUnit.Timeout` to change the timeout value.
https://www.nuget.org/packages/ricaun.RevitTest.TestAdapter/1.4.0-alpha

```C#
[assembly: AssemblyMetadata("NUnit.Timeout", "120")] // Set timeout 120 minutes, default 10 minutes
```

**Comment by ricaun** [comment]
The version `1.4.*` support  `NUnit.Timeout` option to change the timeout value.
```C#
[assembly: AssemblyMetadata("NUnit.Timeout", "120")] // Set timeout 120 minutes, default 10 minutes
```
https://www.nuget.org/packages/ricaun.RevitTest.TestAdapter/1.4.1
